lib/AST/GenerateAST.cpp

```diff
@@ -236,36 +236,68 @@ z3::expr GenerateAST::GetOrCreateEdgeCond(llvm::BasicBlock *from,
   return z_exprs[z_edges[{from, to}]];
 }
 
-z3::expr GenerateAST::GetOrCreateReachingCond(llvm::BasicBlock *block) {
+z3::expr GenerateAST::GetReachingCond(llvm::BasicBlock *block) {
   if (reaching_conds.find(block) == reaching_conds.end()) {
-    if (block->hasNPredecessorsOrMore(1)) {
-      auto cond{z_ctx->bool_val(false)};
-      // Gather reaching conditions from predecessors of the block
-      for (auto pred : llvm::predecessors(block)) {
-        auto has_pred_cond{reaching_conds.find(pred) != reaching_conds.end()};
-        auto edge_cond{GetOrCreateEdgeCond(pred, block)};
-        // Construct reaching condition from `pred` to `block` as
-        // `reach_cond[pred] && edge_cond(pred, block)` or one of
-        // the two if the other one is missing.
-        auto conj_cond{has_pred_cond
-                           ? (z_exprs[reaching_conds[pred]] && edge_cond)
-                           : edge_cond};
-        // Append `conj_cond` to reaching conditions of other
-        // predecessors via an `||`. Use `conj_cond` if there
-        // is no `cond` yet.
-        cond = cond || conj_cond;
-      }
+    return z_ctx->bool_val(false);
+  }
 
-      reaching_conds[block] = z_exprs.size();
-      z_exprs.push_back(cond.simplify());
-    } else {
-      auto cond{z_ctx->bool_val(true)};
+  return z_exprs[reaching_conds[block]];
+}
+
+void GenerateAST::CreateReachingCond(llvm::BasicBlock *block) {
+  auto Prove{[this](z3::expr expr) {
```

Comment:
I have a feeling that we use this piece of code at several places. Is this correct? Can we factor it out into `Utils.cpp` ?

Reply:
Done!